src/Reflection/ClassReflection.php

```diff
@@ -1255,6 +1271,32 @@ public function getPossiblyIncompleteActiveTemplateTypeMap(): TemplateTypeMap
 		return $this->resolvedTemplateTypeMap ?? $this->getTemplateTypeMap();
 	}
 
+	public function getTemplateTypeVarianceMap(): TemplateTypeVarianceMap
```

Comment:
I don't get the need of having these on ClassReflection. Declare-site variance is already taken care of as part of TemplateTag and `ClassReflection::getTemplateTags()`. Call-site variance is taken care of in GenericObjectType. Why do we need this in ClassReflection?

Reply:
Lots of other code deals with `ClassReflection` rather than `GenericObjectType`. As of now, I think the main benefit is in the `getDisplayName()` so that the type displayed in error messages matches the type that's written in code:

```php
<?php

/** @template T */
interface Collection
{
    function clear(int $flags): void;
}

/** @param Collection<*> $collection */
function foo(Collection $collection): void
{
    $collection->clear();
}
```

```
12    Method Collection<*>::clear() invoked with 0 parameters, 1 required.
```

Without the code in ClassReflection, it would say `Collection<mixed>` (or, e.g., `Collection<Animal>` instead of `Collection<covariant Animal>`) which I find rather misleading.

---

I'm not so sure about the other usages. Perhaps the methods are not necessary there and could become private for the sake of _this_ PR, but I'll need public access to them in the subsequent PRs anyway

Reply:
Alright, I get it, but I'd prefer if the method name said something like "projection" or "call site" because right now it's a bit confusing. You can access declaration variance through `TemplateTag::getVariance()` too and `getTemplateTypeVarianceMap` on `ClassReflection` seems like the same thing.

Reply:
Ok, that's fair :)